src/packages/packages.controller.ts

```diff
@@ -0,0 +1,198 @@
+import { 
+  Controller, 
+  Get, 
+  Param, 
+  StreamableFile, 
+  NotFoundException, Res
+} from '@nestjs/common';
+import { EntityManager } from '@mikro-orm/postgresql';
+import { Package } from '../model/package.entity';
+import { Response } from 'express';
+import * as stream from 'stream';
+import { Download } from 'src/model/download.entity';
+import { Version } from 'src/model/version.entity';
+
+@Controller('api/v1/packages')
+export class PackagesController {
+  constructor(private readonly em: EntityManager) {}
+
+  @Get(':name')
+  async getPackageName(@Param('name') name: string, @Param('version') version: string) {
+
+    const pkg = await this.em.findOne(Package, { name }, { populate: ['versions'] });
+    if (!pkg) {
+      throw new NotFoundException(`Package "${name}" not found`);
+    }
+
+    const ver = pkg.versions.getItems();
+
+    if (!ver) {
+      throw new NotFoundException(`Version "${version}" not found for package "${name}"`);
+    }
+
+    return {
+      name: pkg.name,
+      versions: ver,
```

Comment:
probably we need to map it to have only array of strings like `["1.0.1", "1.0.2"]` etc. for now client probably doesn't need more